### ISSUE TITLE
Documenting AWS Lambdas or FaaS

### ISSUE THREAD

**Issue by kbariotis** [esdoc/esdoc#429]
I was wondering how would you approach the documentation of an AWS Lambda function (or anything similar).

The handler would be:
```js
// ./src/createRecord.js
module.exports = (event, context, callback) => {
  // do your thing here using `event` and `context`

  callback({}, { success: true})
  // or `callback(new Error())` for errors
}
```

The above function has multiple ways of being called. Through a REST API, directly invoking it using the AWS SDK, listening to an SNS topic, etc. I have nothing enabled so only direct invoking will work and only a member of the team can invoke it. I want to describe that to ESDoc and include it in the generated documentation, to let other team members know that this function can be invoked directly.

Also, I want to let them know that they don't have to wait for a response since the function will return nothing important.

Also, I want to let them know that they don't have to include a context or a callback param cause those are being handled internally through AWS. I want to define a custom input that isn't necessarily the same as the function's signature.

I wrote the above as a question and also as a proposal in case ESDoc wants to take the route to document the uprising FaaS technology such as AWS Lambdas.

Thank you

**Comment by angelbeltran** [comment]
Why not just provide a description above the lambda (you can use a [@desc](https://esdoc.org/manual/usage/tags.html#-desc) tag), and then describe the parameters with [@param](https://esdoc.org/manual/usage/tags.html#-param) tags? If you are consistent in you documentation, by not setting a [@return](https://esdoc.org/manual/usage/tags.html#-return) tag, you can imply that there is no return. In specifying the context and callback parameters you can point out they are [nullable](https://esdoc.org/manual/usage/tags.html#type-syntax).

I'm not sure what you mean by a "custom input that isn't necessarily the same as the function's signature."

I don't see what additional features ESDoc could or need to provide to document any AWS Lambdas. Aren't they just functions?

**Comment by kbariotis** [comment]
> Aren't they just functions?

No, they are not just functions, meaning they have a different purpose and cannot be treated as regular functions.

One cannot directly call them from a JS module. She can but won't be much useful.

I am thinking of having a way to specify that a function:

* Must be called by other means, direct invocation through AWS-SDK, listens to SNS topic, etc.
* Cannot be called directly
* The function's signature have no meaning to the caller, I have to specify a different input schema

I could, of course, use the regular ESDocs tags to define that but the generated HTML wouldn't be that descriptive. That's why I asked how are you using the regular tags to define that.

**Comment by ozbillwang** [comment]
@kbariotis 

Did you work it out finally?

I have similar request to run `esdoc` on a serverless project (nodejs), but there is nothing generated. 

Any sample codes for me?
